dpe/src/dpe_instance.rs

```diff
@@ -14,29 +14,41 @@ use crate::{
 
 use core::marker::PhantomData;
 
-pub struct DpeInstance<C: Crypto> {
+pub struct DpeInstance<'a, C: Crypto> {
     contexts: [Context; MAX_HANDLES],
     support: Support,
+    localities: &'a [u32],
 
     // All functions/data in C are static and global. For this reason
     // DpeInstance doesn't actually need to hold an instance. The PhantomData
     // is just to make the Crypto trait work.
     phantom: PhantomData<C>,
 }
 
-impl<C: Crypto> DpeInstance<C> {
+impl<C: Crypto> DpeInstance<'_, C> {
     const DEFAULT_CONTEXT_HANDLE: [u8; HANDLE_SIZE] = [0; HANDLE_SIZE];
+    pub const AUTO_INIT_LOCALITY: u32 = 0;
 
-    pub fn new(support: Support) -> DpeInstance<C> {
+    /// Create a new DPE instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `support` - optional functionality the instance supports
+    /// * `localities` - all possible valid localities for the system
+    pub fn new(support: Support, localities: &[u32]) -> DpeInstance<C> {
+        assert!(!localities.is_empty());
```

Comment:
nit: One concern I have is if we panic, it will be pretty hard to debug for SoC integrators. I think I would prefer we surface an error to runtime firmware so it can give some diagnostic information in the status registers.

The list of supported localities will probably ultimately be provided by the SoC integrator (even if it's just part of the Caliptra firmware image metadata)

Reply:
That's fair. I figured this should be caught way before anything gets shipped because it wouldn't even boot. But, debugging those scenarios can be a nightmare. I changed it to return a `Result`.